Bugfix: Crop module should keep aspect if provided

jenshannoschwalm opened the pull request:
Due to missing keep-track-of-selected-aspect the `modify_roi_out()` returned sub-optimal sizes for roi width&height while calculating `dt_dev_pixelpipe_get_dimensions()`.

We might still have non-perfect dimensions
1. Many ratios would require int arithmetics providing "proper" sizes
2. Due to calculating in preview space we might have dimensions one-less than desired

To test old-vs-new behaviour use `-d pipe` and watch out for lines starting with `modify roi OUT` for crop module.

1. Overall leading to better export sizes.
2. @TurboGit the third commit introduces a module version bump to keep old edits for exporting. Personally i think we should not do that but prefer the better new way in all cases though ...

jenshannoschwalm commented:
Part of a better fix than #16692

victoryforce commented:
> 2\. Personally i think we should not do that but prefer the better new way in all cases though ...

I agree absolutely. If we're fixing something, there's no point in keeping worse older edits.

jenshannoschwalm commented on src/iop/crop.c:
```diff
@@ -167,6 +169,40 @@ dt_iop_colorspace_type_t default_colorspace(dt_iop_module_t *self,
+    n->keep_ratio = FALSE;
+
+    *new_params = n;
+    *new_params_size = sizeof(dt_iop_crop_params_t);
```
right of course :-)

jenshannoschwalm commented on src/iop/crop.c:
```diff
@@ -167,6 +169,40 @@ dt_iop_colorspace_type_t default_colorspace(dt_iop_module_t *self,
+    } dt_iop_crop_params_v1_t;
+
+    const dt_iop_crop_params_v1_t *o = (dt_iop_crop_params_v1_t *)old_params;
+    dt_iop_crop_params_t *n = (dt_iop_crop_params_t *)malloc(sizeof(dt_iop_crop_params_t));
```
Sure

jenshannoschwalm commented:
Fixed as required, squashed and force-pushed.

jenshannoschwalm commented:
Don't worry, 5 min work

jenshannoschwalm commented:
That look wrong indeed, at least if you somehow changed the crop size or area, that might fail with the version handling. Will check...

jenshannoschwalm commented:
Ok. Found the culprit for the mentioned 1:1 ratio issue, also added support for "original image ratio". Added some debug log info (verbose mode) to make sure we can check and report issues - if there are. 

As the module version bump is not required i removed that commit, squashed and force-pushed fixed version.

jenshannoschwalm commented:
Release note would be: Fixed some issues related to cropping module and output/export dimensions.

jenshannoschwalm commented:
Fixed building issue.

@ralfbrown @victoryforce @elstoc this got more complicated but i think i have sorted everything out correctly. Would appreciate testing on your side ...

jenshannoschwalm commented:
3rd commit is safe work from imageio.
Mainly code maintenance, more important - have some better guess for `max_possible_scale` avoiding dt crashes because of insane memory consumptions.

ralfbrown commented:
> @ralfbrown @victoryforce @elstoc this got more complicated but i think i have sorted everything out correctly. Would appreciate testing on your side ...

Probably won't have time for testing until middle of next week.
